internal/controller/instaslice_daemonset.go

```diff
@@ -185,130 +175,84 @@ func (r *InstaSliceDaemonsetReconciler) Reconcile(ctx context.Context, req ctrl.
+				// the keep latency low for realizing slices.
+				device, retCodeForDevice := nvml.DeviceGetHandleByUUID(allocations.GPUUUID)
+				if retCodeForDevice != nvml.SUCCESS {
+					log.Error(ret, "error getting GPU device handle")
+				}
+				var giProfileId, ciProfileId int
+				for _, item := range instaslice.Spec.Migplacement {
+					if item.Profile == allocations.Profile {
+						giProfileId = item.Giprofileid
+						ciProfileId = item.Giprofileid
+						break
 					}
+				}
+				giProfileInfo, retCodeForGi := device.GetGpuInstanceProfileInfo(giProfileId)
+				if retCodeForGi != nvml.SUCCESS {
+					log.Error(retCodeForGi, "error getting GPU instance profile info", "giProfileInfo", giProfileInfo, "retCodeForGi", retCodeForGi)
+				}
 
-					log.Info("The profile id is", "giProfileInfo", giProfileInfo.Id, "Memory", giProfileInfo.MemorySizeMB, "pod", allocations.PodUUID)
-					createCiAndGi := true
-					migInfos, err := populateMigDeviceInfos(device)
-					if err != nil {
-						// MIG walking can fail but at this point we are unsure if slices exists
-						// hence we optimistically try to create ci and gi.
-						log.Error(err, "walking MIG devices failed")
-					}
-					// if ci and gi exist, we need to assign those to the respective allocation
-					for migUuid, migDevice := range migInfos {
-						// a (nvidia) GPU can get max of 7 workloads that can have same gi profile info on a GPU
-						// collect such similar profiles and bind it to allocation chosen by controller and add it to cache
-						if migDevice.giInfo.ProfileId == giProfileInfo.Id && migDevice.uuid == allocations.GPUUUID {
-							// search the slice chosen by the controller and add to cache when value is empty or does not exists
-							if _, exists := cachedPreparedMig[allocations.PodUUID]; exists {
-								if allocations.Start == migDevice.start {
-									log.Info("found existing slice for ", "pod", allocations.PodName, "migdevicestart", migDevice.start)
-									cachedPreparedMig[allocations.PodUUID] = preparedMig{gid: migDevice.giInfo.Id, miguuid: migUuid, cid: migDevice.ciInfo.Id}
-									createCiAndGi = false
-									break
-								}
-							}
-						}
+				log.Info("The profile id is", "giProfileInfo", giProfileInfo.Id, "Memory", giProfileInfo.MemorySizeMB, "pod", allocations.PodUUID)
+				createCiAndGi := true
+				createdMigInfos, errCreatingAndPolpulatingMigs = populateMigDeviceInfos(device)
+				if errCreatingAndPolpulatingMigs != nil {
+					// MIG walking can fail but at this point we are unsure if slices exists
+					// hence we optimistically try to create ci and gi.
+					log.Error(errCreatingAndPolpulatingMigs, "walking MIG devices failed")
+				}
+				// if ci and gi exist, we need to assign those to the respective allocation
+				if createCiAndGi {
+					createdMigInfos, errCreatingAndPolpulatingMigs = r.createSliceAndPopulateMigInfos(ctx, device, allocations, giProfileInfo, placement, ciProfileId)
+					if errCreatingAndPolpulatingMigs != nil {
```

Comment:
Going by convention, you can simply use `err` (here and every other place)

Reply:
Thanks, working on it.